modules/common/src/main/scala/notebook/front/widgets/package.scala

```diff
@@ -336,13 +336,20 @@ package object widgets {
     @volatile var currentPoints = points
     @volatile var currentMax = maxPoints
 
-    maxPointsBox.currentData --> Connection.fromObserver { max:Int =>
-      currentMax = max
+    def samplingWarning(maxEntriesLimit: Int): String = {
       if (currentMax >= toPoints.count(currentC)) {
-        warnMax("")
-      } else{
-        warnMax(" (Warning: randomly sampled "+currentMax + " entries)")
+        ""
+      } else {
+        originalData match {
+          case _ @ DataFrame => " (Warning: showing first "+currentMax + " rows)"
```

Comment:
oupsy. dataframes are available in post-df package only... so this will need some horrible indirection.